client/handle_http.go

```diff
@@ -1888,56 +1905,62 @@ func downloadObject(transfer *transferFile) (transferResults TransferResults, er
 	var fileWriter io.Writer
 	if transfer.xferType == transferTypeDownload {
 		var info os.FileInfo
-		if info, err = os.Stat(localPath); err != nil {
-			if os.IsNotExist(err) {
-				// If we're unpacking, the destination must be a directory. Create it directly.
-				if transfer.packOption != "" {
-					if err = os.MkdirAll(localPath, 0700); err != nil {
-						return
-					}
-					if info, err = os.Stat(localPath); err != nil {
-						return
+		// In dry-run mode, skip actual file operations
+		if transfer.job != nil && transfer.job.dryRun {
+			log.Infof("DRY RUN: Would download %s to %s", transfer.remoteURL.Path, localPath)
+			fileWriter = io.Discard
+		} else {
+			if info, err = os.Stat(localPath); err != nil {
+				if os.IsNotExist(err) {
+					// If we're unpacking, the destination must be a directory. Create it directly.
+					if transfer.packOption != "" {
+						if err = os.MkdirAll(localPath, 0700); err != nil {
```

Comment:
I would assume dry-run mode does not create any directories - it shouldn't do any writes to the filesystem!

Instead, just record any directories you would make.

Reply:
You were supposed to remove this `mkdir` call as well (along with any other `mkdir` calls in the patch).